### PR TITLE
[DataGrid] Stop checkbox ripple on blur

### LINKED ISSUE #2998: [DataGrid] Checkbox ripple not consistent when navigating with arrows
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When navigating with arrows on the select column, checkbox's Ripple are not removed when going up

### Expected behavior 🤔

_No response_

### Steps to reproduce 🕹

Steps:

1. Go on [https://mui.com/x/](https://mui.com/x/)
2. click on a checkbox
3. use arrows to go down, everything is normal
4. use arrows to go up, each checkbox keep its Ripple after being blured
5. repeat up and dow to increase the number of ripples

### Context 🔦

_No response_

### Your environment 🌎

Chrome and Firefox

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Linux 5.13 Ubuntu 21.10 21.10 (Impish Indri)
    CPU: (8) x64 Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
    Memory: 3.50 GB / 15.35 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 16.8.0 - ~/.nvm/versions/node/v16.8.0/bin/node
    Yarn: 1.21.1 - /usr/bin/yarn
    npm: 7.21.0 - ~/.nvm/versions/node/v16.8.0/bin/npm
  Managers:
    Apt: 2.3.9 - /usr/bin/apt
    Composer: 1.9.0 - /usr/local/bin/composer
    pip2: 19.3.1 - ~/.local/bin/pip2
    pip3: 20.1.1 - ~/anaconda3/bin/pip3
    RubyGems: 3.2.5 - /usr/bin/gem
  Utilities:
    CMake: 3.18.4 - /usr/bin/cmake
    Make: 4.3 - /usr/bin/make
    GCC: 11.2.0 - /usr/bin/gcc
    Git: 2.32.0 - /usr/bin/git
    FFmpeg: 4.4 - /usr/bin/ffmpeg
  Virtualization:
    Docker: 20.10.8 - /usr/bin/docker
  IDEs:
    Nano: 5.6.1 - /usr/bin/nano
    VSCode: 1.61.2 - /snap/bin/code
    Vim: 8.2 - /usr/bin/vim
  Languages:
    Bash: 5.1.8 - /usr/bin/bash
    Go: 1.14.1 - /usr/local/go/bin/go
    Java: 1.8.0_302 - /usr/bin/javac
    Perl: 5.32.1 - /usr/bin/perl
    PHP: 8.0.8 - /usr/bin/php
    Python: 3.8.3 - /home/alexandre/anaconda3/bin/python
    Python3: 3.8.3 - /home/alexandre/anaconda3/bin/python3
    Ruby: 2.7.4 - /usr/bin/ruby
  Databases:
    SQLite: 3.32.3 - /home/alexandre/anaconda3/bin/sqlite3
  Browsers:
    Chrome: 94.0.4606.61
    Firefox: 93.0
  Monorepos:
    Yarn Workspaces: 1.21.1
    Lerna: 4.0.0

```
</details>

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by m4theushw**
Fixes #2998 

Preview: https://deploy-preview-3835--material-ui-x.netlify.app/components/data-grid/demo/

### CODE REVIEW

**Comment by siriwatknp** [comment]
👋 The [migration PR](https://github.com/mui/mui-x/pull/3730) has been merged.

Please follow these steps to make sure that the contents are all updated. (Sorry for the inconvenience)

1. pull latest master from upstream to your branch
2. if your PR has changes on the `*.md` or demo files, you are likely to encounter conflict because all of them have been moved to the new folder.
   2.1 revert the change on those markdown files you did
   2.2 pull latest master from upstream (you should not see conflict)
   2.3 make the changes again at `docs/data/material/*`
3. run `yarn docs:api`
    - you might see the changes in `docs/pages/material/api/*` if your PR touches some of the components
    - it is okay if there is no changes

If you are struggle with the steps above, feel free to tag @siriwatknp

File: packages/grid/_modules_/grid/components/columnSelection/GridCellCheckboxRenderer.tsx [hunk 1]
```diff
@@ -61,9 +62,12 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridCellPa
...
 
     React.useLayoutEffect(() => {
-      if (hasFocus && checkboxElement.current) {
-        const input = checkboxElement.current.querySelector('input')!;
-        input!.focus();
+      if (hasFocus) {
+        const input = checkboxElement.current?.querySelector('input');
+        input?.focus();
+      } else if (rippleRef.current) {
+        // Only available in @mui/material v5.4.1 or later
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/components/columnSelection/GridCellCheckboxRenderer.tsx:69]: https://github.com/mui-org/material-ui/pull/30901 is only available in this version.

**Review by cherniavskii — APPROVED** [review]
Great work fixing the issue that turned out to be a rabbit hole 😀

**Comment by m4theushw** [comment]
> Should we add a note in changelog to inform users that this bug is solved only for versions of @mui/material larger larger than v5.4.1?

@alexfauquette Maybe. We can add a comment in #3835. Only one user reacted to the bug. #3375 also needs this fix so users will have to update the core version eventually.

**Review by oliviertassinari — COMMENTED** [review]
The peer dependency version looks wrong. The `touchRippleRef` prop was only added in https://github.com/mui/material-ui/releases/tag/v5.4.1 but we allow lower versions:

https://github.com/mui/mui-x/blob/8044100bd9320d3f7e1ff8dd9a0f61e60d58891e/packages/grid/x-data-grid/package.json#L55

This was surfaced in https://mui.zendesk.com/agent/tickets/2771

**Comment by m4theushw** [comment]
@oliviertassinari I can't open the Zendesk ticket: "Access denied". The bug in #2998 only affects users using `checkboxSelection` or the `actions` column. I wouldn't change the peer dependency only because of these features.

**Comment by oliviertassinari** [comment]
> I can't open the Zendesk ticket: "Access denied".

@m4theushw Oops, I have changed the group from Sales to MUI X, it should work now (I can't add you to sales without getting all the emails 🙃). 

> I wouldn't change the peer dependency only because of these features.

From my perspective, the boat has already shipped, the peer dependency in practice is already v5.4.1, proof:

https://codesandbox.io/s/datagriddemo-material-demo-forked-cgqhur?file=/demo.tsx

> Warning: React does not recognize the `touchRippleRef` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `touchrippleref` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

so I think that by updating the "peerDependencies" field, we would improve the DX be triggering a warning .

**Pull request merged**